web3_proxy/src/accepted_currencies.rs

```diff
@@ -0,0 +1,31 @@
+/// Accepted Currencies, their respective ERC20 token function, data and decimals
+
+pub struct Currency {
+    name: String,
+    address: Address,
+    decimals: u8
+}
+
+const CURRENCIES: HashMap<Address, Currency> = lazy_static! {
+    let mut currencies: HashMap<Address, Currency> = HashMap();
+
+    currencies.insert(
+        "0xdAC17F958D2ee523a2206206994597C13D831ec7".parse::<Address>().unwrap(),
```

Comment:
@WyseNynja I suppose we will only have a single currency that we accept?

Reply:
No. Launch will include multiple stablecoins. 

I am pretty sure the factory contract has a list of accepted currencies already. We should query that rather than hard coding here. We don't want to be forced to sync deploys with changes to the smart contracts. 

Also, tokens bridged to other chains have a different address

Reply:
I think we will still need some hard-coded list to convert from absolute token amounts to decimals (i.e. stablecoins can have different decimal points, might be needed for conversion)

Reply:
would probably need something like this: https://github.com/Uniswap/v2-core/blob/master/contracts/UniswapV2Pair.sol#L38 in the payments contract to not have to read them from the storage slots directly